### ISSUE TITLE
calling Keyring from Apache on Windows  KeyError: 'USERPROFILE'\r

### ISSUE THREAD

**Issue by jaraco** [jaraco/keyring#94]
I was getting the following error when calling Keyring from Apache on Windows
 KeyError: 'USERPROFILE'\r

I was able to hack it to work by updating util/platform.py.  See the updated function below.  This is not a general fix, but it allowed me to get past this problem and I wanted to share my solution.

```
#!python
def _data_root_Windows():
    try:
        root = os.environ['LOCALAPPDATA']
    except KeyError:
        **if os.environ.has_key('USERPROFILE'):
            root = os.path.join(os.environ['USERPROFILE'], 'Local Settings')
        else:
            root = os.path.join("C:\Program Files (x86)\Apache Software Foundation\Apache2.2\conf", 'Local Settings')**
    return os.path.join(root, 'Python Keyring')

```

---
- Bitbucket: https://bitbucket.org/kang/python-keyring-lib/issue/94
- Originally reported by: Anonymous
- Originally created at: 2013-02-19T23:31:06.657

**Comment by jaraco** [comment]
An easier workaround might have been to simply set `LOCALAPPDATA="C:\Program Files (x86)\Apache Software Foundation\Apache2.2\conf"` in the environment in which the Apache service runs.

Still, keyring could do better here.

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
In addition to , fall back to , and then to the current directory, when determining the data root on Windows. Fixes #94.

→ <<cset 3a79739d9541>>

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
Detect and support Windows XP explicitly. Ref #94.

→ <<cset 5ee8a6f2c4a0>>

---

Original comment by: Jason R. Coombs
